libraries/AP_Notify/Display.cpp

```diff
@@ -297,11 +297,22 @@ bool Display::init(void)
     // initialise driver
     switch (pNotify->_display_type) {
         case DISPLAY_SSD1306:
-            _driver = new Display_SSD1306_I2C(hal.i2c_mgr->get_device(NOTIFY_DISPLAY_I2C_BUS, NOTIFY_DISPLAY_I2C_ADDR));
+            for(uint8_t i=0; i<3; i++) {
+                _driver = new Display_SSD1306_I2C(hal.i2c_mgr->get_device(i, NOTIFY_DISPLAY_I2C_ADDR));
+                if (_driver && _driver->hw_init()) {
```

Comment:
> However, doing this instance-method-hw_init thing allows you to start building up state in the created object, which is nice.

But even with the probe method, it would still do that. The probe method creates the instance for you, so everything is basically the same, it just happens inside the driver.

> However - not trying to fix everything with this patch, just one thing :-)

I honestly don't think it's a separate issue. You are fixing the issue of being able to use a display on any I2C bus. We can use two methods for that:

- have a new DISPLAY_BUS parameter
- auto-detect the display by looking at all I2C buses

You chose the later, which, in my perspective, implies doing auto-detection in the right way.